### ISSUE TITLE
NET-1534 Rule Idea: `void` method implementations should not be `async`

### ISSUE THREAD

**Issue by Tim-Pohlmann** [SonarSource/sonar-dotnet#9711]
```csharp
interface IInterface
{
    void InterfaceMethod();
}

class Base
{
    protected virtual void BaseMethod() {}
}

class Sample : Base, IInterface
{
    public async void InterfaceMethod() {}           // Noncompliant 
    protected override async void BaseMethod() {}    // Noncompliant
    protected virtual async void VirtualMethod() {}  // Noncompliant
    async void OtherMethod() {}                      // Compliant: covered by S3168
}
```
This rule complements S3168, which ignores methods that implement an interface, override a base method, or are virtual. Having this functionality as a separate rule allows users to tailor their analysis to their needs.

---
Irrelevant Jira import test
https://example.com and http://example.com

**Comment by Tim-Pohlmann** [comment]
The implementation should be a straightforward adaption of S3168.

**Comment by qhris** [comment]
This looks good to me!

Some questions regarding the exceptions raised in https://github.com/SonarSource/rspec/pull/4547:

Would this capture all these scenarios?
* Methods implementing an interface
* Methods overriding a base class method
* Virtual methods

From the wording here it should capture all 3 cases, correct?

**Comment by Tim-Pohlmann** [comment]
I clarified it. Thanks for pointing it out!

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1534](https://sonarsource.atlassian.net/browse/NET-1534)

[NET-1534]: https://sonarsource.atlassian.net/browse/NET-1534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by Corniel** [comment]
Similar to [AsyncFixer03](https://github.com/semihokur/AsyncFixer)
